mpd/mpd.go

```diff
@@ -250,31 +254,70 @@ func (as *AdaptationSet) AddNewContentProtectionRoot(defaultKIDHex string) (*CEN
 	return cp, nil
 }
 
-// Adds a new content protection scheme for Widevine DRM.
-// This does not include the PSSH headers in the Manifest.
-func (as *AdaptationSet) AddNewContentProtectionSchemeWidevine(pssh *string) (*WidevineContentProtection, error) {
-	if pssh != nil {
-		if _, err := base64.StdEncoding.DecodeString(*pssh); err != nil {
-			return nil, err
-		}
+// AddNewContentProtectionSchemeWidevine adds a new content protection scheme for Widevine DRM to the adaptation set.
+// If wvHeader is empty or nil then the ContentProtection element will contain a <cenc:pssh> element which is a base64
+// representation of the PSSH box that would be found in the media init segment.
+// wvHeader - can be at most 1 byte slice
+// !!! Note: this function will accept any wvHeader value !!!
+func (as *AdaptationSet) AddNewContentProtectionSchemeWidevine(wvHeaders ...[]byte) (*WidevineContentProtection, error) {
+	if len(wvHeaders) > 1 {
+		panic("AddNewContentProtectionSchemeWidevine accepts at most 1 wvHeader slice")
 	}
 
-	cp := &WidevineContentProtection{
-		PSSH: pssh,
+	var wvHeader []byte
+	if len(wvHeaders) == 1 {
+		wvHeader = wvHeaders[0]
 	}
-	cp.SchemeIDURI = Strptr(CONTENT_PROTECTION_WIDEVINE_SCHEME_ID)
-	cp.XMLNS = Strptr(CENC_XMLNS)
 
-	err := as.AddContentProtection(cp)
+	cp, err := NewWidevineContentProtection(wvHeader)
+	if err != nil {
+		return nil, err
+	}
+
+	err = as.AddContentProtection(cp)
 	if err != nil {
 		return nil, err
 	}
 	return cp, nil
 }
 
-// Adds a new content protection scheme for PlayReady DRM.
+func NewWidevineContentProtection(wvHeader []byte) (*WidevineContentProtection, error) {
+	cp := &WidevineContentProtection{}
+	cp.SchemeIDURI = Strptr(CONTENT_PROTECTION_WIDEVINE_SCHEME_ID)
+
+	if len(wvHeader) > 0 {
+		cp.XMLNS = Strptr(CENC_XMLNS)
+		wvSystemID, err := hex.DecodeString(CONTENT_PROTECTION_WIDEVINE_SCHEME_HEX)
+		if err != nil {
+			panic(err.Error())
```

Comment:
It means that a constant, defined in the code, that should be a decodable hex string, is not a decodable hex string. I'm okay with panicking.